### PR TITLE
fix: 🐛 subscribe to http driver observables

### PR DESCRIPTION
**Pull request by NachoVazquez**
## PR Checklist

Please check if your PR fulfills the following requirements:

```
[ x ] The commit message follows our guidelines: CONTRIBUTING.md#commit
[ ] Tests for the changes have been added (for bug fixes / features)
[ x ] Docs have been added / updated (for bug fixes / features)
```

## PR Type

What kind of change does this PR introduce?

```
[ x ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

- `HttpDriver` never executes because inner Observables are not being subscribed.
- `HttpClientModule` needs to be manually imported on the client app/library.

Issue Number: N/A

## What is the new behavior?

- `HttpDriver` subscribe to the inner Observable, the driver performs correctly.
- `HttpDriverModule` imports `HttpClientModule` making it an implicit dependency for client apps/libs.

## Does this PR introduce a breaking change?

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Comment by LayZeeDK** [comment]
If we want to import `HttpClientModule`, we need to make several changes because `HttpClient` and related classes are very expensive dependencies when it comes to bundle size. We don't want to impose this on consumers that are not going to use the `HttpDriver`:
1. Put the `HttpDriver` and related classes in a secondary entry point.
1. Add `@angular/common/http` to peer dependencies of the secondary entry point.

Additionally, I think we should make these changes:
1. Create an `HttpDriverRootModule` that imports the `HttpClientModule`.
1. Guard against importing the `HttpDriverModule` directly as only `HttpDriverModule.forRoot` should be used. See `ConsoleDriverModule`.

**Review by LayZeeDK — CHANGES_REQUESTED** [review]
If we want to import `HttpClientModule`, we need to make several changes because `HttpClient` and related classes are very expensive dependencies when it comes to bundle size (~135-150 KB). We don't want to impose this on consumers that are not going to use the `HttpDriver`:
1. Put the `HttpDriver` and related classes in a secondary entry point.
1. Add `@angular/common/http` to peer dependencies of the secondary entry point.

Additionally, I think we should make these changes:
1. Create an `HttpDriverRootModule` that imports the `HttpClientModule`.
1. Guard against importing the `HttpDriverModule` directly as only `HttpDriverModule.forRoot` should be used. See `ConsoleDriverModule`.

You could split this PR into two PRs. One for the bugfix, one for the secondary entry point and NgModule-related changes.

**Comment by LayZeeDK** [comment]
I realize that we're already importing `HttpClient` in the `HttpDriver`, so it doesn't make a difference to remove it from the `HttpDriverModule`. The main point is that this should be in a secondary entrypoint or maybe even a separate package.

File: projects/ngworker/lumberjack/src/lib/log-drivers/http-driver/README.md [hunk 1]
```diff
@@ -110,14 +132,16 @@ The logic of the private `log` method is simple. It stores in an in-memory list
 
 Then the logs are attempted to be send. On success we clear the list of pending logs, otherwise, we keep adding items until the server recovers.
 
+The `sendLogPackage` method has been optimized to run outside **Angular** `ngZone`, avoiding unnecessary changeDetections cycles.
```
Comment (LayZeeDK) [inline projects/ngworker/lumberjack/src/lib/log-drivers/http-driver/README.md:135]:
```suggestion
The `sendLogPackage` method has been optimized to run outside **Angular**'s `NgZone`, avoiding unnecessary change detection cycles.
```

File: projects/ngworker/lumberjack/src/lib/log-drivers/http-driver/http.driver.ts [hunk 2]
```diff
@@ -24,30 +24,30 @@ export class HttpDriver implements LogDriver {
...
+  private log(logEntry: string, logLevel: LumberjackLogLevel): void {
     const { logWagonSize } = this.config;
 
     this.logWagon.push({ logEntry, level: logLevel });
 
     if (this.logWagon.length >= logWagonSize) {
-      return this.sendLogPackage().pipe(tap(() => (this.logWagon = [])));
-    } else {
-      return of();
+      this.sendLogPackage()
```
Comment (LayZeeDK) [inline projects/ngworker/lumberjack/src/lib/log-drivers/http-driver/http.driver.ts:48]: This implementation is pretty poor. We don't keep track of which messages were sent to the server. We just clear all messages after receiving the response. We need to come up with a better solution. But let's leave that for another PR.

**Review by LayZeeDK — CHANGES_REQUESTED** [review]

**Comment by LayZeeDK** [comment]
>Additionally, I think we should make these changes:
>
>1. Create an `HttpDriverRootModule` that imports the `HttpClientModule`.
>1. Guard against importing the `HttpDriverModule` directly as only `HttpDriverModule.forRoot` should be used. See `ConsoleDriverModule`.

We'll work on these in another PR.

File: projects/ngworker/lumberjack/src/lib/log-drivers/http-driver/http.driver.ts (hunk 2 above)
Comment (NachoVazquez) [reply projects/ngworker/lumberjack/src/lib/log-drivers/http-driver/http.driver.ts:53]: I agree, we also have pending moving towards an immutable implementation.

**Review by LayZeeDK — APPROVED** [review]
